flake.nix

```diff
@@ -3,30 +3,40 @@
     nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
     flake-parts.url = "github:hercules-ci/flake-parts";
 
-    purs-nix.url = "github:purs-nix/purs-nix";
+    purs-nix.url = "github:purs-nix/purs-nix/docs-preview";
     purs-nix.inputs.nixpkgs.follows = "nixpkgs";
   };
 
   outputs = inputs@{ self, flake-parts, nixpkgs, ... }:
     flake-parts.lib.mkFlake { inherit self; } {
       systems = [ "x86_64-linux" "x86_64-darwin" ];
       imports = [
+        ./purs-nix.nix
         ./foo
         ./bar
       ];
-      perSystem = { self', system, pkgs, lib, ... }: {
+      perSystem = { self', config, system, pkgs, lib, ... }: {
+        purs-nix = self.inputs.purs-nix {
+          inherit system;
+          overlays =
+            [
+              (self: super: {
+                inherit (self'.packages)
+                  foo bar;
```

Comment:
This just works, and it is great. But `self'.packages.{foo,bar}` come from external module (see ./foo/default.nix for example). Can we inline this? @ursi 

Basically:

```nix
self: super: {
  foo = { src = ./foo; dependencies = ... };
}
```

Once the local packages themselves are defined inside the overlays like this, foo/default.nix for instance can just do `config.purs-nix.ps-pkgs.foo` to get the package. The **problem** however is that we can't define `foo-js` because that requires the `ps` object. Can that be solved perhaps using passthru? Viz.: `config.purs-nix.ps-pkgs.foo.ps.modules.Foo.bundle { ... }`. Or is there an alternative way to achieve this?

Reply:
passthru on the derivation built with `build` is what comes to mind. You could make a new function for building packages that do this.